File: src/data_models/process_model/process_start_response_payload.ts
/**
 * Contains a response from the process engine, which will be send after a process model was started.
 * Depending on the type of start callback used, this will also contain information about the correlation result.
 */
export class ProcessStartResponsePayload {
  /**
   * The id of the correlation in which the started process model runs.
   */
  public correlationId: string;
  /**
   * Optional: If the process engine is to wait for the process instance to finish,
   * this will contain the id of the end event with which process execution was finished.
   */
  public endEventId?: string;
  /**
   * Optional: If the process engine is to wait for the process instance to finish,
   * this will contain the final result with which process execution was finished.
   */
  public tokenPayload?: string;
}

File: dotnet/src/data_models/process_model/ProcessStartResponsePayload.cs
namespace ProcessEngine.ConsumerAPI.Contracts.DataModel
{
    /// <summary>
    /// Contains a response from the process engine, which will be sent after a
    /// ProcessModel was started. Depending on the type of start callback used,
    /// this will also contain information about the Correlation result.
    /// </summary>
    public class ProcessStartResponsePayload
    {
        /// <summary>
        /// The ID of the ProcessInstance.
        /// </summary>
        public string ProcessInstanceId { get; set; }

        /// <summary>
        /// The ID of the Correlation in which the started ProcessModel runs.
        /// </summary>
        public string CorrelationId { get; set; }

        /// <summary>
        /// Optional: If the process engine is set to wait for the
        /// ProcessInstance to finish, this will contain the ID of the EndEvent
        /// with which the ProcessInstance was finished.
        /// </summary>
        public string EndEventId { get; set; }

        /// <summary>
        /// Optional: If the ProcessEngine is set to wait for the
        /// ProcessInstance to finish, this will contain the final result with
        /// which ProcessInstance was finished.
        /// </summary>
        public object TokenPayload { get; set; }
    }
}
